plugins/database/mongodb/mongodb.go

```diff
@@ -218,7 +218,45 @@ func (m *MongoDB) RevokeUser(ctx context.Context, statements dbplugin.Statements
 	return nil
 }
 
-// RotateRootCredentials is not currently supported on MongoDB
+// RotateRootCredentials change the password for the root credentials of vault in the mongoDB
+// Currently not support custom statements
 func (m *MongoDB) RotateRootCredentials(ctx context.Context, statements []string) (map[string]interface{}, error) {
-	return nil, errors.New("root credentaion rotation is not currently implemented in this database secrets engine")
+	m.Lock()
+	defer m.Unlock()
+
+	if len(m.Username) == 0 || len(m.Password) == 0 {
+		return nil, errors.New("username and password are required to rotate")
+	}
+
+	session, err := m.getConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	password, err := m.GeneratePassword()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Rotate statement
+	changeUserPasswordCmd := changeUserPasswordCommand{
+		m.Username,
+		password,
+	}
+
+	// Run Rotate Root credential statement
+	err = session.Run(changeUserPasswordCmd, nil)
```

Comment:
I'm not fully up to speed on MongoDB's security model but it looks like some of the other operations are run in the context of a database.  Is this required for this operation.